### PR TITLE
Improve macOS-HS libvirt XML file.

### PR DESCRIPTION
**Pull request by Ghabry**
- Always enable the tsc timer, as MacOSX depends on this CPU timer as the timer source.

It is documented that MacOS uses the TSC for timing, therefore is better to always enable it by default because it runs too fast otherwise on all CPUs I tested. With this option enabled smbios = 2 can be removed.

- Remove AVX2 because this is not supported on older CPUs 

- Set a tsc frequency to support Suspend to Disk and an explanation how to get the value correct.

- Allow remote spice, otherwise you can't see the video output when you use SSH to connect to a remote VM server.

- Add note about SeLinux

- Added a note about disabling NumLock

### CODE REVIEW

File: macOS-HS-libvirt.xml [hunk 1]
```diff
@@ -66,18 +61,16 @@
...
-
-     Please remember to also change the kvm hidden state as well, as described above.
--->
+  <!-- Note: The frequency is required for KVMs "Suspend to Disk" feature and
+       must match your host CPUs TSC frequency (in Hz).
+       You can obtain the TSC frequency under Linux with "dmesg | grep -i tsc".
+       When you don't need "Suspend to Disk" delete the frequency attribute.
+  -->
+  <clock offset='utc'>
+    <timer name='tsc' frequency='3400022000'/>
```
Comment (kholia) [inline macOS-HS-libvirt.xml]: Where does this hardcoded frequency come from? It can't be right for everybody, or is it?

File: macOS-HS-libvirt.xml [hunk 2]
```diff
@@ -50,9 +46,8 @@
   </os>
   <features>
     <acpi/>
-<!-- Note: On Fedora 27 w/ an i7-5820K CPU and repository QEMU version as of 27-Jan-2018, the guest experienced severe clock issues. If you encounter these issues, try changing this to <hidden state ='off'/> -->
     <kvm>
-      <hidden state='on'/>
+      <hidden state='off'/>
```
Comment (kholia) [inline macOS-HS-libvirt.xml]: Is this change really required or can we keep the former value?

File: macOS-HS-libvirt.xml [hunk 3]
```diff
@@ -66,18 +61,16 @@
     <feature policy='require' name='xsave'/>
     <feature policy='require' name='avx'/>
     <feature policy='require' name='xsaveopt'/>
-    <feature policy='require' name='avx2'/>
```
Comment (kholia) [inline macOS-HS-libvirt.xml]: The default should probably support modern CPUs fully. Folks having older CPUs should change this setting themselves.

File: macOS-HS-libvirt.xml [hunk 4]
```diff
@@ -141,8 +134,8 @@
       <address type='usb' bus='0' port='3'/>
     </input>
     <input type='keyboard' bus='ps2'/>
-    <graphics type='spice'>
-      <listen type='none'/>
+    <graphics type='spice' autoport='yes'>
+      <listen type='address'/>
```
Comment (kholia) [inline macOS-HS-libvirt.xml]: :+1:

File: macOS-HS-libvirt.xml [hunk 5]
```diff
@@ -159,11 +152,10 @@
     </hub>
     <memballoon model='none'/>
   </devices>
+  <!-- Note: Remove the next line when SELinux is not installed -->
```
Comment (kholia) [inline macOS-HS-libvirt.xml]: :+1:

File: macOS-HS-libvirt.xml (hunk 2 above)
Comment (Ghabry) [reply macOS-HS-libvirt.xml]: This change is required for the tsc clock feature

File: macOS-HS-libvirt.xml [hunk 6]
```diff
@@ -9,7 +9,8 @@
 	To install this file, you may place it at ~/.config/libvirt/qemu/
 	and run: virsh define macOS-HS-libvirt.xml.
 
-	This configuration has been tested in Fedora 27 with stock QEMU-KVM.
+	This configuration has been tested in Fedora 27 and Arch Linux with stock
+	QEMU-KVM.
```
Comment (kholia) [inline macOS-HS-libvirt.xml]: Perhaps mention a month + year for Arch Linux.

File: macOS-HS-libvirt.xml (hunk 3 above)
Comment (Ghabry) [reply macOS-HS-libvirt.xml]: okay I readded the line

File: macOS-HS-libvirt.xml (hunk 1 above)
Comment (Ghabry) [reply macOS-HS-libvirt.xml]: See the comment above. The value must match your Host CPU.
I simply inserted the value I currently use, shall I replace it with something else?

File: macOS-HS-libvirt.xml [hunk 7]
```diff
@@ -159,11 +152,10 @@
     </hub>
     <memballoon model='none'/>
   </devices>
+  <!-- Note: Remove the next line when SELinux is not installed -->
   <seclabel type='dynamic' model='selinux' relabel='yes'/>
   <qemu:commandline>
     <qemu:arg value='-device'/>
     <qemu:arg value='isa-applesmc,osk=ourhardworkbythesewordsguardedpleasedontsteal(c)AppleComputerInc'/>
-    <qemu:arg value='-smbios'/>
```
Comment (kholia) [inline macOS-HS-libvirt.xml]: Is this removal strictly required? I am not sure about this.

File: macOS-HS-libvirt.xml [hunk 8]
```diff
@@ -66,18 +61,18 @@
...
-
-     Please remember to also change the kvm hidden state as well, as described above.
--->
+  <clock offset='utc'>
+  <!-- Note: The frequency is required for KVMs "Suspend to Disk" feature and
+       must match your host CPUs TSC frequency (in Hz).
+       You can obtain the TSC frequency under Linux with "dmesg | grep -i tsc".
+       When you don't need "Suspend to Disk" delete the frequency attribute.
+  -->
+    <timer name='tsc' frequency='3400022000'/>
```
Comment (kholia) [inline macOS-HS-libvirt.xml]: Can you please experiment with `-cpu Penryn,kvm=on,vendor=GenuineIntel,+invtsc,vmware-cpuid-freq=on` options?

Once these options are implemented, no manual tsc tweaking should be required.

Comment (kholia) [reply macOS-HS-libvirt.xml]: The `kvm=on` may be required when exposing NVIDIA GPUs. I am not 100% sure about this.

File: macOS-HS-libvirt.xml (hunk 7 above)
Comment (Ghabry) [reply macOS-HS-libvirt.xml]: This was added in https://github.com/kholia/OSX-KVM/pull/93 to resolve incorrect MacOSX performance which is properly resolved with TSC. I can readd it, but I always ran MacOSX without this line and never encountered issues.

File: macOS-HS-libvirt.xml (hunk 8 above)
Comment (wroersma) [reply macOS-HS-libvirt.xml]: does that timer frequency feature add allow snapshots to work?

Comment (kholia) [reply macOS-HS-libvirt.xml]: > Once these options are implemented, no manual tsc tweaking should be required.

@Ghabry Hi, did these options help?

**Comment by kholia** [comment]
These changes have been merged into the repository now.
